HIPs/incubation/waterfall.md

```diff
@@ -0,0 +1,236 @@
+
+The efficiency of the Waterfall network is dependent on successful collaboration between the Coordinating and
+BlockDAG shards, which work in parallel.
+Every system validator has two essential components: the coordinator and the verifier, both playing key roles within
+their respective shards.
+Waterfall implementation provides for the possibility of deploying several autonomous validators on each full node,
+with a common ledger and a pool of transactions.
+Such nodes are infrastructure objects deployed on separate computers (servers) and performing all operations for
+sending and processing data.
+This approach enables Waterfall to achieve scalability without sacrificing decentralization.
+
+![Waterfall Network](../images/waterfall/network.svg "Waterfall Network")
+
+In a BlockDAG shard, all received transactions are first added to its DAG-based ledger and are applied to alter
+the network state only after they are finalized in the Coordinating network since the DAG structure does not have
+a natural ordering.
+A registry of verifiers is responsible for assigning block producers in each slot at the start of each epoch.
+
+The Coordinating network handles the crucial tasks of linearizing (ordering) and finalizing the BlockDAG ledger,
+thereby enhancing security and synchronization across the entire system.
+This network also holds information about the approved blocks generated on the BlockDAG shard.
+Therefore, the execution of transactions and smart contract calls are separated from the consensus protocol.
+
+With Waterfall, a specially developed subnetwork technology ensures scalability in terms of transaction processing
+capacity within the shard networks; in other words, it provides horizontal sharding.
+A transaction pool, the set of valid pending transactions to be recorded in blocks, is split between network verifiers by
+applying a hierarchical and graph-based clustering algorithm.
+All verifiers of the BlockDAG shard have the same ledger and the same general state of processing of all finalized blocks.
+Thus, as the number of verifiers and transactions grows, the number of blocks in each slot will also increase.
+Although, at the moment, subnetworks parallelize the validation of transactions and their inclusion in blocks,
+further processing of blocks to include them in the ledger and change the current state is performed by the nodes of
+all subnetworks.
+
+At present, the network can process [over 10,000 TPS](https://docs.waterfall.network/tests/mainnet-test19), providing major benefits for developers and enterprises.
+This high throughput ensures efficient processing, which is crucial for DeFi, IoT, Web3 gaming, and more.
+Currently, on Waterfall, several projects, including Lightning Works, Indigo Nexus LLC, and WaterSwap DEX, 
+are in the technical documentation stage.
+
+Further parallel processing hierarchical sharding will be intended to decrease the total computational network load and
+reduce volumes of stored data.
```

Comment:
Can you please explain the reduction in stored data volume? Ideally, each shard network verifier replicates the state and syncs the ledger after block finalization by the coordinator nodes.

Reply:
Yes, each verifier execute transactions that have been finalized by all coordinators within a shard.